src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java

```diff
@@ -113,7 +110,7 @@ public String getMechanismName() {
 
     @Override
     public void evaluateRequest(final HttpServerRequest request) throws HttpAuthenticationException {
-        final String realmName = selectRealm(request);
+        final String defaultRealm = selectDefaultRealm();
```

Comment:
We may want to double check the available realms as the mechanism is actually created - at that point we should be able to identify if we have realms available for the mechanism to function or fail to be created if no realms were available.

The previous code only waited until evaluateRequest so we could fallback to using the Host header.